libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -163,6 +168,44 @@ impl<'a> Validator<'a> {
         errors.ok()
     }
 
+    fn validate_names_for_indexes(
+        &self,
+        ast_schema: &ast::SchemaAst,
+        schema: &dml::Datamodel,
+    ) -> Result<(), ErrorCollection> {
+        let mut errors = ErrorCollection::new();
+        let mut index_names = HashSet::new();
+
+        let multiple_indexes_with_same_name_are_supported = match self.source {
+            Some(source) => source.combined_connector.supports_multiple_indexes_with_same_name(),
+            None => false,
+        };
+
+        for model in schema.models() {
+            if let Some(ast_model) = ast_schema.find_model(&model.name) {
+                for index in model.indices.iter() {
+                    if let Some(index_name) = &index.name {
+                        if index_names.contains(index_name) && !multiple_indexes_with_same_name_are_supported {
+                            let ast_index = ast_model
+                                .directives
+                                .iter()
+                                .find(|directive| directive.name.name == "index")
+                                .unwrap();
+                            println!("{:?}", index_name);
```

Comment:
Do we want to use `println` or should we utilize the logger here?

Reply:
Ah that's a leftover from debugging, I'll remove it :)